pkg/stringutils/stringutils.go

```diff
@@ -33,14 +36,49 @@ func GenerateRandomASCIIString(n int) string {
 // Ellipsis truncates a string to fit within maxlen, and appends ellipsis (...).
 // For maxlen of 3 and lower, no ellipsis is appended.
 func Ellipsis(s string, maxlen int) string {
-	r := []rune(s)
-	if len(r) <= maxlen {
+	byteLen := len(s)
+	if byteLen == utf8.RuneCountInString(s) {
+		r := []rune(s)
+		if byteLen <= maxlen {
+			return s
+		}
+		if maxlen <= 3 {
+			return string(r[:maxlen])
+		}
+		return string(r[:maxlen-3]) + "..."
```

Comment:
If we're improving this function, perhaps this is the right moment to use the actual ellipsis (`"…"`) character for this; not sure if that is one or two positions wide, but at least it allows showing more of the original string 👍